cmake/linker/ld/target.cmake

```diff
@@ -77,7 +78,7 @@ macro(configure_linker_script linker_script_gen linker_pass_define)
       COMMAND ${CMAKE_C_COMPILER}
       -x assembler-with-cpp
       ${NOSYSDEF_CFLAG}
-      -MD -MF ${linker_script_gen}.dep -MT ${linker_script_gen}
+      -MD -MF ${linker_script_gen}.dep -MT ${base_name}/${linker_script_gen}
```

Comment:
This was removed here: https://github.com/zephyrproject-rtos/zephyr/commit/a0ba67df63577a91ccd00d6a8c229b74928216e3

Are you perhaps having `CMP0116` set to OLD behavior in your system ?